### PR TITLE
MC-9606

### PR DESCRIPTION
**Pull request by atpw25**

### CODE REVIEW

**Review by gammonpeter — CHANGES_REQUESTED** [review]

**Comment by atpw25** [comment]
All changes have been committed.

File: src/app/term/codeset-list/term-codeset-list.component.html [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    <ng-container matColumnDef="author">
+      <th mat-header-cell *matHeaderCellDef mat-sort-header style="width: 20%;">Author</th>
+      <td mat-cell *matCellDef="let item"> {{item.author}} </td>
+    </ng-container>
+
+    <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
+    <tr mat-row *matRowDef="let row; columns: displayedColumns;"></tr>
+  </table>
+  <div class="mdm--mat-pagination" [ngClass]="{'is-hidden':totalItemCount < 6}">
+    <mat-paginator [length]="totalItemCount" [pageSize]="pageSize" pageIndex="0" showFirstLastButtons></mat-paginator>
```
Comment (gammonpeter) [inline src/app/term/codeset-list/term-codeset-list.component.html:45]: please use mdm-paginator as this handles user configs

File: src/app/term/codeset-list/term-codeset-list.component.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    this.refreshFetchOptions();
+
+    // First data fetch
+    this.codesets.fetchData();
+  }
+
+  refreshFetchOptions() {
+    this.codesets.pageable = {
+      max: this.paginator?.pageSize || this.pageSize,
+      offset: (this.paginator?.pageIndex * this.paginator?.pageSize) || 0
```
Comment (gammonpeter) [inline src/app/term/codeset-list/term-codeset-list.component.ts:99]: using mdm-paginator handles these calculations

File: src/app/terminology/term-list/term-list.component.html [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+          </button>
+        </div>
+      </td>
+    </ng-container>
+
+    <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
+    <tr mat-row *matRowDef="let row; columns: displayedColumns;"></tr>
+  </table>
+  <div class="mdm--mat-pagination" [ngClass]="{'is-hidden':totalItemCount < 6}">
+    <mat-paginator [length]="totalItemCount" [pageSize]="pageSize" pageIndex="0" showFirstLastButtons></mat-paginator>
```
Comment (gammonpeter) [inline src/app/terminology/term-list/term-list.component.html:65]: switch to using mdm-paginator

**Review by gammonpeter — CHANGES_REQUESTED** [review]

**Review by gammonpeter — APPROVED** [review]

**Pull request merged**